internal/logic/logic_test.go

```diff
@@ -120,3 +139,606 @@ func Test_htmlPdfServiceLogic_Ping(t *testing.T) {
+				expected := respModel.Response{
+					Status:  http.StatusInternalServerError,
+					Message: codes.GetErr(codes.ErrReadFileFail),
+					Data:    nil,
+				}
+				if !reflect.DeepEqual(x, &expected) {
+					t.Errorf("want %v got %v", expected, x)
+				}
+			},
+		},
+		{
+			name:        "Failure:: Replace:: SaveFile fail",
+			requestBody: strings.NewReader("abc"),
+			setupFunc: func() *htmlPdfServiceLogic {
+				mockHtmlsvc := mock.NewMockHtmlToPdf(mockCtrl)
+				mockHtmlsvc.EXPECT().GetJsonFromHtml([]byte("abc")).Return([]byte("abc"), nil)
+				mockDatasource := mock.NewMockDataSource(mockCtrl)
+				mockDatasource.EXPECT().SaveFile("1", []byte("abc"), time.Duration(0)).Return(errors.New(""))
+				mockDatasource.EXPECT().GetFile("1").Return([]byte(""), nil)
+				rec := &htmlPdfServiceLogic{
+					dsSvc: mockDatasource,
+					htSvc: mockHtmlsvc,
+				}
+				return rec
+			},
+			validateFunc: func(x *respModel.Response) {
+				expected := respModel.Response{
+					Status:  http.StatusInternalServerError,
+					Message: codes.GetErr(codes.ErrFileStoreFail),
+					Data:    nil,
+				}
+				if !reflect.DeepEqual(x, &expected) {
+					t.Errorf("want %v got %v", expected, x)
+				}
+			},
+		},
+		{
+			name:        "Failure:: Replace:: GetJsonFromHtml fail",
+			requestBody: strings.NewReader("abc"),
+			setupFunc: func() *htmlPdfServiceLogic {
+				mockHtmlsvc := mock.NewMockHtmlToPdf(mockCtrl)
+				mockHtmlsvc.EXPECT().GetJsonFromHtml([]byte("abc")).Return(nil, errors.New(""))
+				mockDatasource := mock.NewMockDataSource(mockCtrl)
+				mockDatasource.EXPECT().GetFile("1").Return([]byte(""), nil)
+				rec := &htmlPdfServiceLogic{
+					dsSvc: mockDatasource,
+					htSvc: mockHtmlsvc,
+				}
+				return rec
+			},
+			validateFunc: func(x *respModel.Response) {
+				expected := respModel.Response{
+					Status:  http.StatusInternalServerError,
+					Message: codes.GetErr(codes.ErrFileConversionFail),
+					Data:    nil,
+				}
+				if !reflect.DeepEqual(x, &expected) {
+					t.Errorf("want %v got %v", expected, x)
+				}
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			rec := tt.setupFunc()
+			resp := rec.Replace("1", tt.requestBody.(io.Reader))
+			tt.validateFunc(resp)
+		})
+	}
+
+}
+
+func Test_HtmlToPdf(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	tests := []struct {
+		name         string
+		requestBody  string
+		setupFunc    func() *htmlPdfServiceLogic
+		validateFunc func(*respModel.Response)
+	}{
+		{
+			name:        "Success:: HtmlToPdf",
+			requestBody: "1",
+			setupFunc: func() *htmlPdfServiceLogic {
+				js, _ := json.Marshal(map[string]interface{}{
+					"Pages": []interface{}{
+						map[string]interface{}{
+							"Base64PageData": base64.StdEncoding.EncodeToString([]byte("abc")),
+						},
+						map[string]interface{}{
+							"Base64PageData": nil,
+						},
+					},
+				})
+				var a uint8
+				a = 10
+				js = append(js, a)
```

Comment:
It was demonstrated to only compare the data that is expected by unmarshaling/decoding to revert any characters that may have been added for formatting and only compare the data.
Change this.

Reply:
done